File: XPile 3-17/Programs/test_builtins.js

function double_it(x) {
    return (x * 2);
}

function is_even(x) {
    return ((x % 2) === 0);
}

function main() {
    let nums = [1, 2, 3, 4, 5];
    let doubled = nums.map(double_it);
    for (const d of doubled) {
        console.log(d);
    }
    let evens = nums.filter(is_even);
    for (const e of evens) {
        console.log(e);
    }
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: XPile 3-17/Programs/test_builtins.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def double_it(x):
    return (x * 2)

def is_even(x):
    return ((x % 2) == 0)

def main():
    nums = [1, 2, 3, 4, 5]
    doubled = list(map(double_it, nums))
    for d in doubled:
        print(d)
    evens = list(filter(is_even, nums))
    for e in evens:
        print(e)
    return 0


if __name__ == "__main__":
    main()
